nbgrader/apps/zipcollectapp.py

```diff
@@ -0,0 +1,486 @@
+        'extracted',
+        help=dedent(
+            """
+            The name of the directory that contains assignment submission files
+            extracted or copied from the `archive_directory`. This corresponds
+            to the `collect_step` variable in the `collect_structure` config
+            option.
+            """
+        )
+    ).tag(config=True)
+
+    timezone = Unicode(
+        "UTC",
+        help="Timezone for recording timestamps"
+    ).tag(config=True)
+
+    timestamp_format = Unicode(
+        "%Y-%m-%d %H:%M:%S %Z",
+        help="Format string for timestamps"
+    ).tag(config=True)
+
+    extractor_plugin = Type(
+        ExtractorPlugin,
+        klass=BasePlugin,
+        help=dedent(
+            """
+            The plugin class for extracting the archive files in the
+            `archive_directory`.
+            """
+        )
+    ).tag(config=True)
+
+    collector_plugin = Type(
+        FileNameCollectorPlugin,
+        klass=BasePlugin,
+        help=dedent(
+            """
+            The plugin class for processing the submitted file names after
+            they have been extracted into the `extracted_directory`.
+            """
+        )
+    ).tag(config=True)
+
+    collector_plugin_inst = Instance(FileNameCollectorPlugin).tag(config=False)
+    extractor_plugin_inst = Instance(ExtractorPlugin).tag(config=False)
+
+    @default("classes")
+    def _classes_default(self):
+        classes = super(ZipCollectApp, self)._classes_default()
+        classes.append(ExtractorPlugin)
+        classes.append(ZipCollectApp)
+        classes.append(self.collector_plugin)
+        return classes
+
+    def _format_collect_path(self, collect_step):
+        kwargs = dict(
+            downloaded=self.downloaded_directory,
+            assignment_id=self.assignment_id,
+            collect_step=collect_step,
+        )
+
+        path = os.path.join(
+            self.course_directory,
+            self.collect_directory_structure,
+        ).format(**kwargs)
+
+        return path
+
+    def _mkdirs_if_missing(self, path):
+        if not check_directory(path, write=True, execute=True):
+            self.log.warn("Directory not found. Creating: {}".format(path))
+            os.makedirs(path)
+
+    def _clear_existing_files(self, path):
+        if not os.listdir(path):
+            return
+
+        if self.force:
+            self.log.warn("Clearing existing files in {}".format(path))
+            rmtree(path)
+            os.makedirs(path)
+        else:
+            self.fail(
+                "Directory not empty: {}\nuse the --force option to clear "
+                "previously existing files".format(path)
+            )
+
+    def _find_student(self, info):
+        try:
+            with open_gradebook(self.db_url) as gradebook:
+                return gradebook.find_student(info['student_id'])
+        except MissingEntry:
+            return None
+
+    def get_timestamp(self):
```

Comment:
I think here you want to get the datetime in UTC format, because when you parse the timestamp below (line 371) it is converted into UTC format without a tzinfo. So by making it have a tzinfo here, you will later get an error like this:

```
TypeError: can't compare offset-naive and offset-aware datetimes
```